### DISCUSSION TITLE
Version 1.10.0rc2

### DISCUSSION THREAD

**Discussion by jpivarski** [scikit-hep/awkward#1684] [New releases]
## Bug-fixes

* fix: allow string arrays to merge without unions by @agoose77 in https://github.com/scikit-hep/awkward/pull/1671
* fix: carry parameters through broadcasting by @agoose77 in https://github.com/scikit-hep/awkward/pull/1679

## Other

* refactor: move categorical functions to operations by @agoose77 in https://github.com/scikit-hep/awkward/pull/1674
* chore: remove `dev/build-awkward.sh` by @agoose77 in https://github.com/scikit-hep/awkward/pull/1675

**Full Changelog**: https://github.com/scikit-hep/awkward/compare/v1.10.0rc1...v1.10.0rc2

<hr /><em>This discussion was created from the release <a href='https://github.com/scikit-hep/awkward/releases/tag/v1.10.0rc2'>Version 1.10.0rc2</a>.</em>

**Comment by jpivarski** [comment]
@ivirshup and @grst, this release _should_ have everything you need. When it's available (~4 hours), can you test it?

https://github.com/scikit-hep/awkward/actions/runs/3009821080

If not, we'll do an rc3 to try to get it stable before 1.10.0, which is also the last minor number before we split git branches into a v2-only and a v1-legacy branch (https://github.com/scikit-hep/awkward/issues/1626). After that, 1.10.x will be taking v1 bug-fixes, and we'll be dismantling v1 from `main` to be able to release 2.0.0rc1. You'll eventually be moving to 2.0.0rcX, which should only require replacing

```python
import awkward._v2 as ak
```

with

```python
import awkward as ak
```

though there will likely be a time-gap between releasing 1.10.0 and 2.0.0rc1, as we handle all the issues of dismantling v1 from `main`. We'd be able to put bug-fixes into 1.10.1, etc., but hopefully just v1 bug-fixes, not v2 bug-fixes. (Because v2 bug-fixes would have to be merged into both `main` and `main-v1`.)

So that's why we want to ensure that 1.10.0 has everything you need, so that you're not waiting for a v2 bug-fix while we're stuck with a still-transitioning `main` branch.

**Reply by ivirshup** [reply]
Do you have a timeframe in mind for `1.10.0`?

**Reply by jpivarski** [reply]
If you don't see any issues with it, I'll make it today.

On our side, we would _like_ to make it quickly because we're holding off on merging longer-term PRs so that this release consists of only dropping Python 3.6 and bug-fixes for you.

**Reply by jpivarski** [reply]
We're looking at #1688 right now.

**Reply by ivirshup** [reply]
Do you have a timeframe for 2.0?

If you'd like to make the 1.10 release ASAP, I'd be inclined to say don't worry too much about us since we're aiming for the v2 api anyways.

I am not sure I am going to be able to get the anndata PR basically done until at least this weekend.

There are a two major things to change there which involve aligning types and trying a different approach with the behaviors.

**Reply by jpivarski** [reply]
No, we don't need the AnnData PR to be done—that would be a big ask, considering how long it's been open and how much work there is in it. What we're trying to do is to get Awkward 1.10.0 to a plateau of usability so that it will be okay for us to not be able to give you bug-fix releases for a week or so while we do our infrastructure work.

Right now, we will be including #1689, #1676, #1673, and #1562 in 1.10.0rc3 (not all of these are motivated by AnnData, some are older). That is our current best estimate of what will need to go into 1.10.0.

**Reply by jpivarski** [reply]
In the end, only #1676 was held back. I'm starting the 1.10.0rc3 release now.

https://github.com/scikit-hep/awkward/actions/runs/3017192412

**Reply by jpivarski** [reply]
[awkward 1.10.0rc3](https://pypi.org/project/awkward/1.10.0rc3/) is ready to be pip-installed and tested. If that's stable enough for you for (let's say) the next two or three weeks, then we'll make a final release.

After we've dismantled v1 from `main`, we'll be in a position to provide bug-fixes in the 2.0.0rcX series, which do `import awkward as ak` for v2.

**Reply by ivirshup** [reply]
Sorry for the late response here, had a bit of a surprise conference.

The 1.10.0rc3 looks to only decrease errors on our test suite! Not blocked from our side at the moment

**Reply by jpivarski** [reply]
[awkward 1.10.0rc4](https://pypi.org/project/awkward/1.10.0rc4/) is ready for testing. If you have no new issues by tomorrow, I'll release 1.10.0.

@agoose77 has had a lot of success with the v1-removal so far (#1690), so maybe it won't be much of a gap between 1.10.0 and 2.0.0rc1 after all.

**Reply by jpivarski** [reply]
Unless there's any objection, I'm going to make the final 1.10.0 today, and then we'll start dismantling v1 from the `main` branch (meaning: get #1690 working and then apply it).

**Reply by jpivarski** [reply]
I'm going to release 1.10.0 (no rc) now.

**Reply by jpivarski** [reply]
(It's done.)
